src/datasets/arrow_dataset.py

```diff
@@ -4656,6 +4661,10 @@ def to_dict(self, batch_size: Optional[int] = None, batched: bool = False) -> Un
                 indices=self._indices if self._indices is not None else None,
             ).to_pydict()
         else:
+            warnings.warn(
+                "'batched' was deprecated. Use `Dataset.iter(batch_size=batch_size)` instead.",
+                FutureWarning,
+            )
```

Comment:
```suggestion
            if batched != "deprecated":
                warnings.warn(
                    "'batched' was deprecated in version 2.11.0 and will be removed in version 3.0.0. Use `.iter(batch_size=batch_size)` followed by `.to_dict()` on the individual batches instead.",
                    FutureWarning,
                )
            else:
                batched = False 
```
We should have this warning at the beginning of the function. Plus, the default value of `batched` (in the function's signature) should be `"deprecated"`